### PR TITLE
Fix docs for shallow water design doc

### PR DESCRIPTION
**Pull request by xylar**
Docs appear not to have been built locally in #28, so there are several issues to fix.

Checklist
* [x] Documentation:
  * [x] Design document has been generated and added to the docs
  * [x] Documentation has been [built locally](https://e3sm-project.github.io/Omega/develop/devGuide/BuildDocs.html) and changes look as expected

### CODE REVIEW

**Comment by xylar** [comment]
@mark-petersen, I missed a few things in #28 
* Please keep the checklist as part of any PRs you make in the future.  It would have helped catch these issues
* The design doc wasn't added to the `index.md` file. (A few others aren't in alphabetical order as we would like them to be, so I'm fixing that here, too.)
* The link at the top of the file wasn't changed from the template

Unfortunately, the link in the checklist to how to build the docs was broken (also fixed here).  But for future reference, please follow those instructions for any PR that changes the docs to catch these issues, then check that particular item in the checklist.

**Comment by xylar** [comment]
@mark-petersen, feel free to make further changes here if you need to or merge this if you're happy.

File: components/omega/doc/design/ShallowWaterOmega0.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(omega-design-my-class)=
+(omega-design-shallow-water-omega0)=
 # ShallowWaterOmega0
```
Comment (xylar) [inline components/omega/doc/design/ShallowWaterOmega0.md:2]: BTW, I'm not sure how useful this class name is.  Are we going to make such a class or is this just a placeholder?

Comment (xylar) [reply components/omega/doc/design/ShallowWaterOmega0.md:2]: If this isn't a real class name, maybe:
```suggestion
# Omega v0: Shallow Water
```
or something like that?

Comment (mark-petersen) [reply components/omega/doc/design/ShallowWaterOmega0.md:2]: I copied that from the template, but I admit I don't know what it does, or what a class name in markdown is. But just following the pattern, your suggestion is fine.

**Review by mark-petersen — APPROVED** [review]
Thanks, these all look good. I was able to compile the html and view it in a browser. I'm on board with the testing procedure now. Sorry for the inconvenience.

File: components/omega/doc/design/ShallowWaterOmega0.md (hunk 1 above)
Comment (xylar) [reply components/omega/doc/design/ShallowWaterOmega0.md]: This refers to a C++ class name being documented, not a feature of markdown or the docs themselves. I think we should clarify this in the template. Do you want to add that to this branch, @mark-petersen?

**Pull request merged**